### PR TITLE
refactor(lanelet2_map_preprocessor): rework parameter

### PR DESCRIPTION
**Pull request by Motsu-san**
## Description

Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371.

- Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.
  - This PR.
- Create the yaml configs in this package ~~and the autoware_launch repository~~.
  - This PR.
- Create the schema.
  - This PR.
- Update `Parameters` chapter in the readme file.
  - N/A

### Check if the parameters set
```
ros2 launch lanelet2_map_preprocessor transform_maps.launch.xml llt_map_path:="$HOME/autoware_map/sample-map-rosbag/lanelet2_map.osm" pcd_map_path:="$HOME/autoware_map/sample-map-rosbag/pointcloud_map.pcd" llt_output_path:=lanelet2_map_tr.osm pcd_output_path:=pointcloud_map_tr.pcd z:=0.0
[INFO] [launch]: All log files can be found below /home/[your home dir]/.ros/log/2024-02-01-15-47-32-339799-dpc2302001-43645
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [transform_maps-1]: process started with pid [43646]
[transform_maps-1] transforming maps with following parameters
[transform_maps-1] x 0
[transform_maps-1] y 0
[transform_maps-1] z 0
[transform_maps-1] roll 0
[transform_maps-1] pitch 0
[transform_maps-1] yaw 0
[transform_maps-1] Loaded Lanelet2 map
[transform_maps-1] Loaded 326867 data points.
[transform_maps-1] using mgrs grid: 54SUE
[INFO] [transform_maps-1]: process has finished cleanly [pid 43646]
```

### Check if json schema works
1. Confirm if check-jsonschema is installed.
```
pip3 install -U check-jsonschema
```
2. Command below
```
cd [Your Autoware.universe dir]/map/util/lanelet2_map_preprocessor
find -wholename '*/schema/*.schema.json' -printf '%p: ' -execdir bash -c 'check-jsonschema -v --schemafile "$1" ../config/"${1:2:-12}"*.param.yaml' bash '{}' +
```
result
```
map/util/lanelet2_map_preprocessor$ find -wholename '*/schema/*.schema.json' -printf '%p: ' -execdir bash -c 'check-jsonschema -v --schemafile "$1" ../config/"${1:2:-12}"*.param.yaml' bash '{}' +
./schema/transform_maps.schema.json: ok -- validation done
The following files were checked:
  ../config/transform_maps.param.yaml
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

- New Feature: Added support for installing YAML configuration files along with the package.
- Bug Fix: Removed default values for parameters in the launch file and source code, allowing all parameter values to be passed from YAML configuration files.
- Refactor: Improved modularity and maintainability by separating configuration files from the codebase.
- Documentation: Updated release notes to provide a summary of changes made to the package.

### CODE REVIEW

File: map/util/lanelet2_map_preprocessor/launch/transform_maps.launch.xml [hunk 1]
```diff
@@ -4,23 +4,13 @@
...
-  <arg name="roll" default="0.0"/>
-  <arg name="pitch" default="0.0"/>
-  <arg name="yaw" default="0.0"/>
+  <arg name="param_file" default="$(find-pkg-share lanelet2_map_preprocessor)/config/transform_maps.param.yaml"/>
 
   <node pkg="lanelet2_map_preprocessor" exec="transform_maps" name="transform_maps" output="screen">
     <param name="llt_map_path" value="$(var llt_map_path)"/>
     <param name="pcd_map_path" value="$(var pcd_map_path)"/>
     <param name="llt_output_path" value="$(var llt_output_path)"/>
     <param name="pcd_output_path" value="$(var pcd_output_path)"/>
```
Comment (mitsudome-r) [inline map/util/lanelet2_map_preprocessor/launch/transform_maps.launch.xml:13]: I think we can move these paths into the yaml file as well.

Comment (Motsu-san) [reply map/util/lanelet2_map_preprocessor/launch/transform_maps.launch.xml]: @mitsudome-r Thank you. I moved these parameters to param.yaml. Could you review it?

**Comment by Motsu-san** [comment]
@mitsudome-r Could you review this pull-request again?

**Comment by Motsu-san** [comment]
@mitsudome-r Friendly ping.

File: map/util/lanelet2_map_preprocessor/schema/transform_maps.schema.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Parameters for Transforming Maps",
+  "type": "object",
+  "definitions": {
+    "transform_maps": {
+      "type": "object",
+      "properties": {
+        "x": {
```
Comment (mitsudome-r) [inline map/util/lanelet2_map_preprocessor/schema/transform_maps.schema.json:9]: There should be schema for map paths as well:

```
        "llt_map_path": {
          "type": "string",
          "description": "Path pointing to the input lanelet2 file",
          "default": ""
        },
        "pcd_map_path": {
          "type": "string",
          "description": "Path pointing to the input point cloud file",
          "default": ""
        },
        "llt_output_path": {
          "type": "string",
          "description": "Path pointing to the output lanelet2 file",
          "default": ""
        },
        "pcd_output_path": {
          "type": "string",
          "description": "Path pointing to the output point cloud file",
          "default": ""
        },
```

Comment (Motsu-san) [reply map/util/lanelet2_map_preprocessor/schema/transform_maps.schema.json:29]: I've modified it as you said. Thank you.

**Comment by Motsu-san** [comment]
@mitsudome-r I've modified it to be reviewed. Could you review this pull-request again?

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**